src/index.ts

```diff
@@ -0,0 +1,168 @@
+import * as _ from 'underscore';
+import { AWSProvider, ServerlessInstance, ServerlessConfig } from './types';
+
+interface PluginConfig {
+  stages?: string[];
+  roleArn?: string;
+  filterPattern: string;
+  normalizedFilterID: boolean;
+  createLambdaPermission: boolean;
+  destinationARN: string;
+}
+
+const configDefaults = {
+  filterPattern: '',
+  normalizedFilterID: true,
+  createLambdaPermission: true,
+};
+
+interface CFObject<TProps> {
+  Type: string;
+  DependsOn?: string[];
+  Properties: TProps;
+}
+
+interface LambdaPermissionProps {
+  Action: string;
+  Principal: string;
+  FunctionName: string;
+}
+
+interface SubscriptionFilterProps {
+  DestinationArn: string;
+  FilterPattern: string;
+  LogGroupName: string;
+  RoleArn?: string;
+}
+
+type LambdaPermission = CFObject<LambdaPermissionProps>;
+type SubscriptionFilter = CFObject<SubscriptionFilterProps>;
+
+function getStage(options: ServerlessConfig, sls: ServerlessInstance) {
```

Comment:
This function I would suggest making as a part of the class or create a separate file like `utils.ts` to declare such helper functions

Reply:
addressed